tfhe/src/integer/server_key/comparator.rs

```diff
@@ -1689,4 +2266,512 @@ mod tests {
+                        params,
+                        num_tests,
+                        |comparator, lhs, rhs| comparator.[<smart_scalar_ $comparison_name _parallelized>](lhs, rhs),
+                        |lhs, rhs| U256::from(<U256>::$comparison_name(&lhs, &rhs) as u128),
+                    )
+                }
+
+                fn [<scalar_ $comparison_name _parallelized_256_bits>](params:  crate::shortint::ClassicPBSParameters) {
+                    let num_tests = 1;
+                    test_default_scalar_function(
+                        params,
+                        num_tests,
+                        |comparator, lhs, rhs| comparator.[<scalar_ $comparison_name _parallelized>](lhs, rhs),
+                        |lhs, rhs| U256::from(<U256>::$comparison_name(&lhs, &rhs) as u128),
+                    )
+                }
+
+                create_parametrized_test!([<unchecked_scalar_ $comparison_name _parallelized_256_bits>]
+                {
+                    PARAM_MESSAGE_2_CARRY_2,
+                    PARAM_MESSAGE_3_CARRY_3,
+                    PARAM_MESSAGE_4_CARRY_4
+                });
+
+                create_parametrized_test!([<smart_scalar_ $comparison_name _parallelized_256_bits>]
+                {
+                    PARAM_MESSAGE_2_CARRY_2,
+                    // We don't use PARAM_MESSAGE_3_CARRY_3,
+                    // as smart test might overflow values
+                    // and when using 3_3 to represent 256 we actually have more than 256 bits
+                    // of message so the overflow behaviour is not the same, leading to false negatives
+                    PARAM_MESSAGE_4_CARRY_4
+                });
+
+                create_parametrized_test!([<scalar_ $comparison_name _parallelized_256_bits>]
+                {
+                    PARAM_MESSAGE_2_CARRY_2,
+                    // We don't use PARAM_MESSAGE_3_CARRY_3,
+                    // as default test might overflow values
+                    // and when using 3_3 to represent 256 we actually have more than 256 bits
+                    // of message so the overflow behaviour is not the same, leading to false negatives
+                    PARAM_MESSAGE_4_CARRY_4
+                });
+            }
+        };
+    }
+
+    define_scalar_comparison_test_functions!(eq);
+    define_scalar_comparison_test_functions!(ne);
+    define_scalar_comparison_test_functions!(lt);
+    define_scalar_comparison_test_functions!(le);
+    define_scalar_comparison_test_functions!(gt);
+    define_scalar_comparison_test_functions!(ge);
+
+    fn test_unchecked_scalar_min_parallelized_256_bits(
+        params: crate::shortint::ClassicPBSParameters,
+        num_tests: usize,
+    ) {
+        test_unchecked_function(
+            params,
+            num_tests,
+            |comparator, lhs, rhs| comparator.unchecked_min_parallelized(lhs, rhs),
+            std::cmp::min,
+        )
+    }
```

Comment:
I thought these function used to be macro-ized, I'm guessing it was less interesting here to use macros ?

Reply:
For the non scalar test they were not macro-ized, an for the scalar case I just  copy pasted what existed 😬

Reply:
do you think it's worth macro-izing it ? I don't want you to spend too much time on this PR, the functionality is there

Reply:
Hum, i'd say its fine to leave them as is, the code for them is not too long, and I cannot use the same macro as for the ge,gt,lt test as U256::ge/gt,etc take `&U256 as inputs and min/max take plain U256

Reply:
ok !